x-pack/plugins/apm/common/environment_filter_values.ts

```diff
@@ -37,7 +37,7 @@ export const ENVIRONMENT_NOT_DEFINED = {
   text: getEnvironmentLabel(ENVIRONMENT_NOT_DEFINED_VALUE),
 };
 
-export function getEnvironmentEsField(environment: string) {
+export function getEnvironmentEsField(environment?: string) {
```

Comment:
Is this absolutely necessary to make it optional? I assume it was intentional?

Reply:
It's needed because I got environment from the urlParams (https://github.com/elastic/kibana/pull/105132/files/ff5cb47892c23538f95710c6276ad478374a89e4#diff-77218f3284ffc1701914ac54e6a511978c342b54783fc17fc6ce824bacab77afR40) and it is optional, and I thought it wouldn't harm since we already check if it is defined here https://github.com/elastic/kibana/pull/105132/files/ff5cb47892c23538f95710c6276ad478374a89e4#diff-efa89150f46c68fed243a3929945621713f6f32f17658d3c997b3cc5adf80c37R42

Reply:
What about checking before calling the fn? As a rule of thumb: required arguments are better than optional. 
Unless the majority of callers world have to implement the check I don't think this should be made optional.

Reply:
This function is mostly called in the server, here is the first time we'll call it from the client, where environment might be undefined. I can check it before calling it, but this function returns an empty object when environment is not defined, if we decide to change the return value by any other value than an empty object, we'd have to find where this function is called and make sure it also uses the correct new value. 

But anyway, for now, it's okay to have `environment ?  getEnvironmentEsField(environment): {}`  since it's called in one single place but I'm worried that this will start spreading around the code over time.

Reply:
I think you can call `getEnvironmentEsField(ENVIRONMENT_NOT_DEFINED_VALUE)`. We should probably start doing that in other places too.